### DISCUSSION TITLE
How do you keep Code snipet sync between Gitbook and Github

### DISCUSSION THREAD

**Discussion by jayonsoftware** [GitbookIO/community#252] [0. General bugs and questions]
The sample code we have in our doc gets changed on Gitbook by tech editors and also gets changed by developers on GitHub. Right now, it's a cut-and-paste operation, and this manual process introduces a lot of errors. Any thoughts on how to improve this?

**Comment by addisonschultz** [comment]
Hey there! Have you seen the GitHub files integration? It should be helpful for keeping code files in sync https://www.gitbook.com/integrations/github-files

**Reply by jayonsoftware** [reply]
The issue is the embed does not support code coloring. Here is one suggestion https://github.com/orgs/GitbookIO/discussions/231

**Reply by addisonschultz** [reply]
The integration should highlight code correctly! But to make sure I understand, you would like the feature of denoting (with comments) in your code, the parts of the file that should be included/picked up by the integration?

**Reply by bcdurak** [reply]
I am having a similar issue as @jayonsoftware. 

The GitHub files integration provides a great way to display various code snippets in the documentation, however, as the color coding does not work in its current state, it makes them much harder to read.

**Reply by addisonschultz** [reply]
We are tracking that feature request here: https://github.com/GitbookIO/integrations/issues/178

I would love to hop on this if ew have prioritise some other things going on at the moment!

**Comment by addisonschultz** [comment]
Hey @jayonsoftware @bcdurak! I've updated and published some changes to the GitHub and GitLab Files integrations :) Please let me know if there's anything else I missed, but this feature request should now be complete 😄 

Pull Request: https://github.com/GitbookIO/integrations/pull/194

**Reply by bcdurak** [reply]
The code coloring seems to be working now. Thank you for the quick response 👍

**Reply by jayonsoftware** [reply]
I am trying to embed the following 

https://github.com/jayonsoftware/GpuTest/blob/main/Program.cs

But I keep seeing this error.
![Screenshot](https://github.com/GitbookIO/community/assets/1304943/39cb2a51-e0fd-47f8-b516-0ccae7789ed9)

**Reply by addisonschultz** [reply]
Hey @jayonsoftware another question on that - is this happening only with .cs files?

**Reply by jayonsoftware** [reply]
@addisonschultz  Take a look at https://app.gitbook.com/o/EuLCMlHj0R7FTjYXa51D/s/XDOIQrO9OLY3gnurAuGT/readme/demo . First is a java file and syntex coloring is not working, 2nd is a C# file I got those char on the first line, 3rd I am giving the line numbers but that's not working either.

**Reply by addisonschultz** [reply]
Thanks for putting together this demo - there's obviously a good amount of improvements/fixes we'll need to look into. 

Would you mind filing an issue in this repository for the issues you're running into? That will allow myself and my team to priortize and hop into these fixes as we're able to! 

https://github.com/GitbookIO/integrations/issues

**Comment by jayonsoftware** [comment]
Also ideally I am looking for a way to link code snippets that are part of a large code base. for example, in code I can make a start and end region  for example, see my https://doc.apexsharp.com/readme/apexsharp-salesforce-c-api

**Reply by addisonschultz** [reply]
So, we don't have the support right now for start/end blocks - but this integration does support code lines. For example, if you pasted this code link: https://github.com/jayonsoftware/GpuTest/blob/main/Program.cs#L9-L39


You would only see the highlighted lines

**Reply by jayonsoftware** [reply]
Code lines are a bad idea as code changes.
